common/sensor/sensor.c

```diff
@@ -57,189 +72,267 @@ bool access_check(uint8_t sensor_num)
-			sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache = sensor_fail;
+			sensor_config[sensor_config_index_map[sensor_num]].cache = SENSOR_FAIL;
 			printf("sensor[%x] cache read fail status %x\n", sensor_num,
-			       sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache_status);
-			return sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache_status;
+			       sensor_config[sensor_config_index_map[sensor_num]].cache_status);
+			return sensor_config[sensor_config_index_map[sensor_num]].cache_status;
 		}
 	}
 
-	return SNR_UNSPECIFIED_ERROR; // should not reach here
+	return SENSOR_UNSPECIFIED_ERROR; // should not reach here
 }
 
-void disable_snr_poll()
+void disable_sensor_poll()
 {
-	snr_poll_eanble_flag = 0;
+	sensor_poll_enable_flag = false;
 }
 
-void enable_snr_poll()
+void enable_sensor_poll()
 {
-	snr_poll_eanble_flag = 1;
+	sensor_poll_enable_flag = true;
 }
 
-void SNR_poll_handler(void *arug0, void *arug1, void *arug2)
+void sensor_poll_handler(void *arug0, void *arug1, void *arug2)
 {
 	uint8_t poll_num;
-	int reading, SNR_POLL_INTERVEL_ms;
+	int sensor_poll_interval_ms;
+	float reading;
 	k_msleep(1000); // delay 1 second to wait for drivers ready before start sensor polling
 
-	pal_set_sensor_poll_interval(&SNR_POLL_INTERVEL_ms);
+	pal_set_sensor_poll_interval(&sensor_poll_interval_ms);
 
 	while (1) {
 		for (poll_num = 0; poll_num < SENSOR_NUM_MAX; poll_num++) {
-			if (snr_poll_eanble_flag == 0) { /* skip if disable sesnor poll */
+			if (sensor_poll_enable_flag == 0) { /* skip if disable sensor poll */
 				break;
 			}
-			if (SnrNum_SnrCfg_map[poll_num] == sensor_null) { // sensor not exist
+			if (sensor_config_index_map[poll_num] == SENSOR_NULL) { // sensor not exist
 				continue;
 			}
-			get_sensor_reading(poll_num, &reading, get_from_sensor);
+			get_sensor_reading(poll_num, &reading, GET_FROM_SENSOR);
 
 			k_yield();
 		}
-		k_msleep(SNR_POLL_INTERVEL_ms);
+		k_msleep(sensor_poll_interval_ms);
 	}
 }
 
+__weak void pal_set_sensor_poll_interval(int *interval_ms)
+{
+	*interval_ms = 1000;
+	return;
+}
+
+__weak void pal_fix_sensor_config(void)
+{
+	return;
+}
+
+bool stby_access(uint8_t sensor_number)
+{
+	return 1;
+}
+
+bool DC_access(uint8_t sensor_number)
+{
+	return get_DC_on_delayed_status();
+}
+
+bool post_access(uint8_t sensor_number)
+{
+	return get_post_status();
+}
+
 void sensor_poll_init()
 {
 	k_thread_create(&sensor_poll, sensor_poll_stack, K_THREAD_STACK_SIZEOF(sensor_poll_stack),
-			SNR_poll_handler, NULL, NULL, NULL, CONFIG_MAIN_THREAD_PRIORITY, 0,
+			sensor_poll_handler, NULL, NULL, NULL, CONFIG_MAIN_THREAD_PRIORITY, 0,
 			K_NO_WAIT);
 	k_thread_name_set(&sensor_poll, "sensor_poll");
 	return;
 }
 
+uint8_t get_sensor_config_index(uint8_t sensor_num)
+{
+	uint8_t i, j;
+	for (i = 0; i < SENSOR_NUM_MAX; i++) {
+		for (j = 0; j < sensor_config_num; ++j) {
+			if (sensor_num == sensor_config[j].num) {
+				return j;
+			} else if (i == sensor_config_num) {
+				return SENSOR_NUM_MAX;
+			}
```

Comment:
Need else statement

Reply:
Hi @GoldenBug,
We have fixed this and done the forced update.
Please review it. Thanks.